src/test/kotlin/oop/proxy/ScreenVirtualProxyTest.kt

```diff
@@ -0,0 +1,41 @@
+package oop.proxy
+
+import org.hamcrest.core.Is.`is`
+import org.junit.Assert.*
+import org.junit.Test
+
+class ScreenVirtualProxyTest {
+
+  @Test
+  fun `not create real object in proxy construction`() {
+    var screenVirtualProxy = ScreenVirtualProxy({ RealScreen() })
```

Comment:
This is what I mean. Even if the function has not executed yet, you have instanciated the `RealScreen()` object so you can return it later.